artemis-commons/src/main/java/org/apache/activemq/artemis/utils/ActiveMQThreadPoolExecutor.java

```diff
@@ -64,20 +63,15 @@ public void setExecutor(ActiveMQThreadPoolExecutor executor) {
       public boolean offer(Runnable runnable) {
          boolean retval = false;
 
-         // Need to lock for 2 reasons:
-         // 1. to safely handle poll timeouts
-         // 2. to protect the delta from parallel updates
-         synchronized (lock) {
-            if ((executor.getPoolSize() >= executor.getMaximumPoolSize()) || (threadTaskDelta > 0)) {
-               // A new task will be added to the queue if the maximum number of threads has been reached
-               // or if the delta is > 0, which means that there are enough idle threads.
+         if (threadTaskDelta > 0 || (executor.getPoolSize() >= executor.getMaximumPoolSize())) {
```

Comment:
Not a significan change, but why you've reverted the checks here?
The common path should be that you have all the threads busy up and running (so `executor.getPoolSize() >= executor.getMaximumPoolSize() = true`): it would avoid checking the delta...

Reply:
getPoolSize uses a lock underneath in the executor, so we possible can avoid that when we have idle threads.

if threadTaskDelta > 0 (which means we have idle) then we can offer, we only need to see if we are at max if we dont have idle.

Reply:
Ahhh is see that there is a lock there, that's why!